performance-tests/perf-tests.sh

```diff
@@ -59,6 +72,53 @@ warmUp() {
   consumerPerf "$1" "$2" "${WARM_UP_NUM_RECORDS_PRE_TEST}" /dev/null > /dev/null
 }
 
+setupAsyncProfilerKroxy() {
+  sudo sysctl kernel.perf_event_paranoid=1
+  sudo sysctl kernel.kptr_restrict=0
+
+  curl --create-dirs --output-dir /tmp/asprof -O "https://repo1.maven.org/maven2/me/bechberger/ap-loader-all/3.0-9/ap-loader-all-3.0-9.jar"
+
+  mkdir -p /tmp/asprof-extracted
+
+  unzip /tmp/asprof/ap-loader-all-3.0-9.jar -d /tmp/asprof-extracted
+}
+
+deleteAsyncProfilerKroxy() {
+  rm -rf /tmp/asprof
+  rm -rf /tmp/asprof-extracted
+}
+
+startAsyncProfilerKroxy() {
+
+  echo -e "${PURPLE}Starting async profiler${NOCOLOR}"
+
+  docker exec -it ${KROXYLICIOUS_CONTAINER_ID} mkdir -p /root/.local/share/me.bechberger.ap-loader/3.0/bin/../lib/
+
+  docker cp /tmp/asprof-extracted/libs/libasyncProfiler-3.0-linux-arm64.so ${KROXYLICIOUS_CONTAINER_ID}:/root/.local/share/me.bechberger.ap-loader/3.0/bin/../lib/libasyncProfiler.so
+
+  docker exec -it ${KROXYLICIOUS_CONTAINER_ID} ls /root/.local/share/me.bechberger.ap-loader/3.0/lib/
+
+  docker exec -it ${KROXYLICIOUS_CONTAINER_ID} realpath /root/.local/share/me.bechberger.ap-loader/3.0/bin/../lib/libasyncProfiler.so
```

Comment:
this is failing for me with 
```
Running warm up
KROXYLICIOUS_CONTAINER_ID set to b6fbd19e4fab
Starting async profiler
Successfully copied 339kB to b6fbd19e4fab:/root/.local/share/me.bechberger.ap-loader/3.0/bin/../lib/libasyncProfiler.so
libasyncProfiler.so
/root/.local/share/me.bechberger.ap-loader/3.0/lib/libasyncProfiler.so

Failed to change credentials to match the target process: Operation not permitted
++ docker compose -f /home/sam/development/kroxylicious/performance-tests/docker-compose.yaml down
```

Reply:
Same as https://github.com/kroxylicious/kroxylicious/pull/1181#discussion_r1592554877